src/OrleansRuntime/Silo/SiloControl.cs

```diff
@@ -155,28 +171,33 @@ public Task<int> GetActivationCount()
 
         public Task<object> SendControlCommandToProvider(string providerTypeFullName, string providerName, int command, object arg)
         {
-            IReadOnlyCollection<IProvider> allProviders = silo.AllSiloProviders;
-            IProvider provider = allProviders.FirstOrDefault(pr => pr.GetType().FullName.Equals(providerTypeFullName) && pr.Name.Equals(providerName));
+            IProvider provider = null;
+            foreach (var providerManager in this.providerManagers)
+            {
+                try
+                {
+                    var candidate = providerManager.GetProvider(providerName);
+                    if (candidate?.GetType()?.FullName?.Equals(providerTypeFullName) == true)
```

Comment:
better to use string.Equals here and avoid the need to compare an expression to `true`.